### DISCUSSION TITLE
About the estimate Command

### DISCUSSION THREAD

**Discussion by adhranneto** [IRIS-Solutions-Team/IRIS-Toolbox#406] [Q&A]
Hello. I am trying to perform Bayesian estimation and maximum likelihood estimation using the estimate command. Using the same data and the same model, there is a difference in results when using the old version and the latest version of the command. Apparently, the old version of the command performed the optimizations correctly. I might be mistaken, but is there any significant difference in the optimization process or parameterization, or something that I am not noticing?

For the older estimate command, the maximum likelihood estimates of the parameters are:
a1 = 0.23
a2 = 0.065
a3 = 0.58
std shock = 1.34

and, moreover, the values are located at the lowest point of the likelihood function. (I plot the neighborhood). In the new estimate command, this does not happen. The parameter estimates, besides being different, are far from the minimum of the likelihood function.

When performing Bayesian estimation, in the old command, the mode of the posterior distribution coincides with that calculated by the Metropolis-Hastings algorithm. In the new command, this coincidence does not occur. The values are very disparate.

Thank you very much in advance for your attention.
[estimatemodel.txt](https://github.com/user-attachments/files/16014502/estimatemodel.txt)
[kalm_his.csv](https://github.com/user-attachments/files/16014504/kalm_his.csv)
[model.model.txt](https://github.com/user-attachments/files/16014505/model.model.txt)
[myzif.txt](https://github.com/user-attachments/files/16014507/myzif.txt)
[readmodel.txt](https://github.com/user-attachments/files/16014510/readmodel.txt)

**Comment by jaromir-benes** [comment]
Before we go deeper, could you please extract the following pieces of code:

* the *verbatim* lines of code as used with the older iris that set up the
estim struct (initial, lower, upper, prior)
* the *verbatim* line of code. as used with the older iris to run the
estimate command

Then, the same for what you use with the newer version iris.

Thanks
Jaromir

**Reply by adhranneto** [reply]
Here you go!

The file old.txt contains the code for the old version of the maximum likelihood estimation, Bayesian, and Metropolis-Hastings (arwm) commands.
The file new.txt contains the new version.
Just to note: My colleague, who has an older version of MATLAB, ran the old version and got a result. I, with a more recent version of MATLAB, got a different result. I took his code, adapted the syntax of the commands according to the instructions provided on this GitHub, and carefully observed the step-by-step functioning of the algorithm in the new version.

Thank you for your attention and availability.

[new.txt](https://github.com/user-attachments/files/16113284/new.txt)
[old.txt](https://github.com/user-attachments/files/16113285/old.txt)
